src/main/java/ploopy/TaskList.java

```diff
@@ -0,0 +1,172 @@
+     * and rewrites all tasks to storage file.
+     *
+     * @param taskIndex Index of task to be marked done.
+     * @throws PloopyException If a storage file error occurs.
+     */
+    public void markTask(int taskIndex) throws PloopyException {
+        Task current = taskList.get(taskIndex - 1);
+        current.markDone();
+        ui.markTaskMessage(current);
+        storage.rewriteFile(taskList);
+    }
+
+    /**
+     * Marks the provided task as undone, prints the acknowledgement
+     * and rewrites all tasks to storage file.
+     * @param taskIndex Index of task to be marked undone.
+     * @throws PloopyException If a storage file error occurs.
+     */
+    public void unmarkTask(int taskIndex) throws PloopyException {
+        Task current = taskList.get(taskIndex - 1);
+        current.unmark();
+        ui.unmarkTaskMessage(current);
+        storage.rewriteFile(taskList);
+    }
+
+    /**
+     * Deletes the provided task, prints the acknowledgement
+     * and rewrites all tasks to storage file.
+     *
+     *
+     * @param taskNumber Index of task to be marked undone.
+     * @throws PloopyException If a storage file error occurs.
+     */
+
+    public void deleteTask(int taskNumber) throws PloopyException {
+        ui.deleteTaskMessage(taskList.get(taskNumber - 1), taskList.size() - 1);
+        taskList.remove(taskNumber - 1);
+        storage.rewriteFile(taskList);
+    }
+
+    /**
+     * Creates a ToDo task and adds it to the taskList.
+     * Prints acknowledgement and writes task to storage file.
+     *
+     * @param input Name of task.
+     * @throws PloopyException If a storage file error occurs.
+     */
+    public void createToDo(String input) throws PloopyException {
+        Task newTask = new ToDo(input);
+        taskList.add(newTask);
+        ui.addTaskMessage(newTask, taskList.size());
+        storage.writeToFile(newTask);
+    }
+
+    /**
+     * Creates a Deadline task and adds it to the taskList.
+     * Prints acknowledgement and writes task to storage file.
+     *
+     * @param name Name of task.
+     * @param date Date of task.
+     * @throws PloopyException If a storage file error occurs.
+     */
+    public void createDeadline(String name, String date) throws PloopyException {
+        Task newTask = new Deadline(name, date);
+        taskList.add(newTask);
+        ui.addTaskMessage(newTask, taskList.size());
+        storage.writeToFile(newTask);
+    }
+
+    /**
+     * Creates an Event task and adds it to the taskList.
+     * Prints acknowledgement and writes task to storage file.
+     *
+     * @param name Name of task
+     * @param date Date of task
+     * @throws PloopyException If a storage file error occurs.
+     */
+    public void createEvent(String name, String date) throws PloopyException {
+        Task newTask = new Event(name, date);
+        taskList.add(newTask);
+        ui.addTaskMessage(newTask, taskList.size());
+        storage.writeToFile(newTask);
```

Comment:
This code seems to be repeated multiple times in this class, perhaps there is a better way to structure this.